examples/contact-sensor-app/linux/main.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <AppMain.h>
+
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/clusters/network-commissioning/network-commissioning.h>
+#include <app/server/Server.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/Linux/NetworkCommissioningDriver.h>
+
+#if defined(CHIP_IMGUI_ENABLED) && CHIP_IMGUI_ENABLED
+#include <imgui_ui/ui.h>
+#include <imgui_ui/windows/boolean_state.h>
+#include <imgui_ui/windows/occupancy_sensing.h>
+#include <imgui_ui/windows/qrcode.h>
+#endif
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+
+#if CHIP_DEVICE_CONFIG_ENABLE_WPA
```

Comment:
As a followup, this should match all-clusters-app or bridge-app, so it works on Mac too, not just Linux, and doesn't violate the spec on Linux when CHIP_DEVICE_CONFIG_ENABLE_WPA is false.

And whatever app you copied this code from should get fixed too.  ;)

Reply:
```
rg CHIP_DEVICE_CONFIG_ENABLE_WPA examples --files-with-matches | grep linux
examples/thermostat/linux/main.cpp
examples/platform/nxp/se05x/linux/AppMain.cpp
examples/platform/linux/Options.cpp
examples/platform/linux/AppMain.cpp
examples/lighting-app/linux/main.cpp
examples/all-clusters-minimal-app/linux/main-common.cpp
examples/lock-app/linux/main.cpp
examples/ota-requestor-app/linux/main.cpp
examples/contact-sensor-app/linux/main.cpp
```

not sure what I copied this from, probably some of the smaller examples (lock?).
Which app is correct? all-clusters?

Reply:
Updated from all-clusters main-common, however this is surprisingly complex. We should centralize things (and I am unsure if main common is correct .. it does some fifo bits as well and other logic, which is probably not shareable).

Reply:
bridge-app is correct as well, and a bit simpler than all-clusters in that it does not try to support dual wifi+thread devices....